### PR TITLE
Remove the deprecated OMX and MMAL HWA

### PR DESCRIPTION
**Pull request by nyanmisaka**
**Changes**
- Remove the deprecated OMX and MMAL HWA

Web https://github.com/jellyfin/jellyfin-web/pull/3482

### CODE REVIEW

File: MediaBrowser.MediaEncoding/Encoder/EncoderValidator.cs [hunk 1]
```diff
@@ -44,18 +44,7 @@ public class EncoderValidator
...
-            "mpeg2_mmal",
-            "mpeg4_mmal",
-            "vc1_mmal",
-            "h264_opencl",
-            "hevc_opencl",
-            "mpeg2_opencl",
-            "mpeg4_opencl",
-            "vp8_opencl",
-            "vp9_opencl",
-            "vc1_opencl"
```
Comment (Shadowghost) [inline MediaBrowser.MediaEncoding/Encoder/EncoderValidator.cs:58]: Why remove them too?

Comment (nyanmisaka) [reply MediaBrowser.MediaEncoding/Encoder/EncoderValidator.cs:58]: Actually these is no opencl decoder in ffmpeg. I don't know why they were added.
https://github.com/FFmpeg/FFmpeg/search?q=h264_opencl

**Review by Shadowghost — APPROVED** [review]

**Comment by crobibero** [comment]
If the user has this HWA type enabled and we remove it, playback will fail. I suggest we just fallback to "no" hardware encoding.
Note that the dropdown in web will display nothing if the user had OMX/MediaCodec selected

```patch
Index: Jellyfin.Api/Helpers/TranscodingJobHelper.cs
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/Jellyfin.Api/Helpers/TranscodingJobHelper.cs b/Jellyfin.Api/Helpers/TranscodingJobHelper.cs
--- a/Jellyfin.Api/Helpers/TranscodingJobHelper.cs	(revision fab5f37e0e4775eb1d2c80b9d5fa3d98f5be12a6)
+++ b/Jellyfin.Api/Helpers/TranscodingJobHelper.cs	(date 1646495045523)
@@ -458,9 +458,12 @@
                 var audioCodec = state.ActualOutputAudioCodec;
                 var videoCodec = state.ActualOutputVideoCodec;
                 var hardwareAccelerationTypeString = _serverConfigurationManager.GetEncodingOptions().HardwareAccelerationType;
-                HardwareEncodingType? hardwareAccelerationType = string.IsNullOrEmpty(hardwareAccelerationTypeString)
-                    ? null
-                    : (HardwareEncodingType)Enum.Parse(typeof(HardwareEncodingType), hardwareAccelerationTypeString, true);
+                HardwareEncodingType? hardwareAccelerationType = null;
+                if (!string.IsNullOrEmpty(hardwareAccelerationTypeString)
+                    && Enum.TryParse<HardwareEncodingType>(hardwareAccelerationTypeString, out var parsedHardwareAccelerationType))
+                {
+                    hardwareAccelerationType = parsedHardwareAccelerationType;
+                }
 
                 _sessionManager.ReportTranscodingInfo(deviceId, new TranscodingInfo
                 {
```

**Comment by nyanmisaka** [comment]
Encoder will fallback to the default libx264/5 if the selected HWA if not existed in the map.
https://github.com/jellyfin/jellyfin/blob/fab5f37e0e4775eb1d2c80b9d5fa3d98f5be12a6/MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs#L90-L100

**Review by crobibero — APPROVED** [review]

**Review by 1337joe — APPROVED** [review]

**Pull request merged**